From 9d6c904523acce6fa750b0ac3aed3e30674c573f Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sat, 22 May 2021 22:42:36 +0300
Subject: [PATCH] [SW-43025] fwupd: secure i2c update fix2

Following rebases, the module_fwupd_send_status_fast() function gained a
security related "open" functionality.
This forces an update on the "secure i2c update" patch to skip the open
for secured interfaces.

Change-Id: I712f6a87333f789b8b54ee2234296a558d177b47
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/preboot/module_fwupd.c b/preboot/module_fwupd.c
index 9836dac..6f2fba5 100644
--- a/preboot/module_fwupd.c
+++ b/preboot/module_fwupd.c
@@ -133,7 +133,7 @@
 	struct hl_upd_status status;
 	uintptr_t off = (uintptr_t)offset;
 
-	if (offset)
+	if (state->use_security && offset)
 		preboot_security_dynamic_img_sram_open();
 
 	status.val = 0;
